flow/Trace.cpp

```diff
@@ -931,7 +931,9 @@ TraceEvent::~TraceEvent() {
 	} catch( Error &e ) {
 		TraceEvent(SevError, "TraceEventDestructorError").error(e,true);
 		delete tmpEventMetric;
-		throw;
+		// TODO: it is illegal in C++11 to throw an exception in a destructor
+		// so we need to figure out what to do here
+		//throw;
```

Comment:
Just commenting throw here will result with `delete tmpEventMetric;` called two times here. So we need to figure out what to do here before merging this PR.

Reply:
I missed that. we can remove the delete statement but I'm not sure whether there's any code relying on this to throw

Reply:
Me neither. Just started to compile this thing and ran into problems with cmath too. I plan to dig into the code today and try to understand what this `TraceEvent` is used for, also what functions called in this destructor may throw and why, and update you if I will be sure what to do on this line (just for your information, not to stop you from doing the same, as I may fail at this task tbh).

Reply:
TraceEvent is used for logging. Each object creates one log-line in the trace file when destructed. As the logging happens during destruction it makes sense that this is the part that can generate an exception

Reply:
I think marking this destructor as `noexcept(false)` here and in `Trace.h` will suffice for purpose of this PR.

Reply:
And I would also leave `// TODO: try to get rid of throwing exception from destructor.`

Reply:
My initial thought is that the implemented change to log and swallow the error seems reasonable. I'm not sure it's necessary that a failure to write the log here needs to bubble out as an error.

One thing that I wonder about though (and which was possible before) is that if all events are failing with errors for some reason, then logging an event in the catch block is going to get us into a recursive loop.